Coordinate.h

```diff
@@ -17,14 +25,24 @@ class CellCoordinate;
 class NodeCoordinate : public Coordinate {
   public:
     NodeCoordinate(int x, int y) : Coordinate(x, y){};
-    NodeCoordinate operator+(Direction);
+    NodeCoordinate(Direction d) : Coordinate(d){};
```

Comment:
Set as `explicit` to prevent implicit casting from `Direction` to `NodeCoordinate`. This helps prevent hard to detect bugs.